.changeset/empty-clocks-rest.md

```diff
@@ -0,0 +1,16 @@
+---
+'xstate': minor
+---
+
+An array of guards can now be specified in a transition's `cond: ...` property. All guards in the array must pass ("and" behavior) for the transition to be taken.
```

Comment:
That was one of my first feature requests to XState (see https://github.com/statelyai/xstate/issues/414)! I think though that I agree with your past self, this is ambiguous because it's not obvious if this represents AND or OR. 

Since this is a somewhat addressed issue in v5 - is it worth introducing this in v4?

Reply:
One thing I realized is that OR can be represented by separate transitions:

```js
EVENT: [
  { target: 'next', cond: 'thisOne' },
  { target: 'next', cond: 'orThatOne' }
]
```

So maybe it's not ambiguous after all.

I'm still not 100% sure if this should be a core feature, as opposed to `and(...)`, but I put it up here so we can talk about it.

Reply:
> One thing I realized is that OR can be represented by separate transitions:

This is what I've initially responded in that old issue: https://github.com/statelyai/xstate/issues/414#issuecomment-480838103 🤣 

But you don't necessarily have them side by side to look at and compare so when looked at in isolation - this isn't that obvious. Both use arrays, the "nesting level" is different and this makes a difference - I'm somewhat worried that this ain't beginner-friendly. This might not be obvious even for seasoned devs, or a tired dev :P - how many times we have been looking at parenthesis or curly braces put at the wrong place? How many times we have been rechecking operator precedence in non-basic scenarios?

I'm not saying that I'm super opposed to this - I'm just raising concerns. Also - if we are introducing those "operands" in v5 I think this would get deprecared/removed there anyway, in favor of `and(['cond1', 'cond2'])`. So the question is - is it even worth it to introduce a temporary syntax?